CLOUD_MIGRATION_PLAN.md

```diff
@@ -0,0 +1,913 @@
+        echo "  add [size]           - Add new worker (default: s-1vcpu-2gb)"
+        echo "  remove <id>          - Remove worker (graceful shutdown)"
+        echo "  resize <id> <size>   - Resize worker"
+        echo "  list                 - List all workers"
+        echo ""
+        echo "Sizes:"
+        echo "  s-1vcpu-1gb  - 1GB RAM  (\$6/month)"
+        echo "  s-1vcpu-2gb  - 2GB RAM  (\$12/month)"
+        echo "  s-2vcpu-4gb  - 4GB RAM  (\$24/month)"
+        echo "  s-4vcpu-8gb  - 8GB RAM  (\$48/month)"
+        exit 1
+        ;;
+esac
+```
+
+**Worker Image Creation** (`deploy/create_image.sh`):
+
+```bash
+#!/bin/bash
+# Create a snapshot image with Rust and fold pre-installed
+# This speeds up worker startup significantly
+
+# Create base droplet
+doctl compute droplet create "fold-base-$(date +%s)" \
+    --image ubuntu-22-04-x64 \
+    --size s-1vcpu-1gb \
+    --region nyc3 \
+    --wait
+
+DROPLET_ID=$(doctl compute droplet list --format ID,Name --no-header | grep fold-base | awk '{print $1}')
+DROPLET_IP=$(doctl compute droplet get "$DROPLET_ID" --format PublicIPv4 --no-header)
+
+# Wait for SSH to be ready
+sleep 30
+
+# Install dependencies
+ssh "root@$DROPLET_IP" << 'EOF'
+apt-get update
+apt-get install -y build-essential curl git
+
+# Install Rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+source $HOME/.cargo/env
+
+# Clone and build fold
+git clone https://github.com/michaelrauh/fold.git /opt/fold
+cd /opt/fold
+cargo build --release
+
+# Create worker script
+mkdir -p /opt/fold/deploy
+EOF
+
+scp deploy/worker.sh "root@$DROPLET_IP:/opt/fold/deploy/"
+
+# Power off and snapshot
+doctl compute droplet-action power-off "$DROPLET_ID" --wait
+doctl compute droplet-action snapshot "$DROPLET_ID" --snapshot-name "fold-worker-snapshot" --wait
+
+# Cleanup base droplet
+doctl compute droplet delete "$DROPLET_ID" --force
+
+echo "Created snapshot: fold-worker-snapshot"
+```
+
+## Handling Duplicate Work
+
+Since workers don't coordinate, the same input might be processed multiple times. This is handled at the merge phase:
+
+### Merge Strategy
+
+```rust
+// Archives from the same input file can be identified by name prefix
+// e.g., file1_1699900000.tar.zst, file1_1699900001.tar.zst
+
+fn merge_duplicate_archives(archives: Vec<Archive>) -> Archive {
+    // Group by input file
+    let grouped = group_by_input_file(archives);
+    
+    // For each group, keep the one with highest ortho count
+    // (or merge if they have different results)
+    grouped.into_iter()
+        .map(|(input, archives)| {
+            archives.into_iter()
+                .max_by_key(|a| a.ortho_count)
+                .unwrap()
+        })
+        .collect()
+}
+```
+
+### Eventual Consistency
+
+- Multiple workers may process the same file
+- All results are valid (deterministic processing)
+- Merge phase picks the best or combines results
+- **Duplicate work cost < coordination overhead**
```

Comment:
Duplicate work should be avoided by moving active work to a different s3 location